tartufo/config.py

```diff
@@ -279,25 +262,12 @@ def compile_rules(patterns: Iterable[Union[str, Dict[str, str]]]) -> List[Rule]:
                 Rule(
                     name=pattern.get("reason", None),  # type: ignore[union-attr]
                     pattern=re.compile(pattern["pattern"]),  # type: ignore[index]
-                    path_pattern=re.compile(pattern.get("path-pattern", ".*")),  # type: ignore[union-attr]
-                    re_match_type="search",
+                    path_pattern=re.compile(pattern.get("path-pattern", "")),  # type: ignore[union-attr]
+                    re_match_type=pattern.get("match-type", MatchType.Search),  # type: ignore[union-attr]
+                    re_match_scope=pattern.get("scope", Scope.Line),  # type: ignore[union-attr]
```

Comment:
We should ensure that these are valid values. You could do something like this...

```python
try:
    match_type = MatchType(pattern.get("match-type", MatchType.Search.value))
except ValueError as exc:
    raise ConfigException(f"Invalid value for match-type: {pattern.get('match-type')}") from exc
```
Of course, that breaks the list comprehension... but we do want to be sure that they don't supply something weird like `match-type = "ford-prefect"`, or just typo like `match-type = "saerch"`.

Reply:
If I add checks for but `MatchType` and `Scope` here, I can avoid checking for values for `Scope` in `scanner.py`, right?